skbeam/core/tests/test_correlation.py

```diff
@@ -64,22 +66,66 @@ def setup():
 def test_lazy_vs_original():
     setup()
     # run the correlation on the full stack
-    full_gen = lazy_multi_tau(
+    full_gen_one = lazy_one_time(
         img_stack, num_levels, num_bufs, rois)
-    for gen_result in full_gen:
+    for gen_result_one in full_gen_one:
         pass
 
-    g2, lag_steps = multi_tau_auto_corr(num_levels, num_bufs, rois, img_stack)
+    g2, lag_steps = multi_tau_auto_corr(num_levels, num_bufs,
+                                        rois, img_stack)
 
-    assert np.all(g2 == gen_result.g2)
-    assert np.all(lag_steps == gen_result.lag_steps)
+    assert np.all(g2 == gen_result_one.g2)
+    assert np.all(lag_steps == gen_result_one.lag_steps)
+
+    full_gen_two = lazy_two_time(rois, img_stack, stack_size,
+                                 num_bufs, num_levels)
+    for gen_result_two in full_gen_two:
+        pass
+    final_gen_result_two = two_time_state_to_results(gen_result_two)
```

Comment:
To make these variables clearer, rename `gen_result_two` to `gen_state_two` and `final_gen_result_two` to `gen_result_two`